support/k8s/k8s-workload-registrar/controllers/spiffeid_controller.go

```diff
@@ -0,0 +1,295 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/idutil"
+	"github.com/spiffe/spire/proto/spire/api/registration"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/zeebo/errs"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	spiffeidv1beta1 "github.com/spiffe/spire/api/spiffecrd/v1beta1"
+)
+
+// SpiffeIDReconcilerConfig holds the config passed in when creating the reconciler
+type SpiffeIDReconcilerConfig struct {
+	Log          logrus.FieldLogger
+	R            registration.RegistrationClient
+	TrustDomain  string
+	Cluster      string
+	Mgr          ctrl.Manager
+}
+
+// SpiffeIDReconciler holds the runtime configuration and state of this controller
+type SpiffeIDReconciler struct {
+	client.Client
+	c                  SpiffeIDReconcilerConfig
+	myId               *string
+	spiffeIDCollection map[string]string
```

Comment:
This state seems important for proper cleanup. What happens if the registrar is restarted? How does this behave in the face of multiple registrars?

Reply:
Removed this in memory map